tests/testthat/test-register.R

```diff
@@ -614,12 +614,18 @@ extern \"C\" attribute_visible void R_init_testPkg(DllInfo* dll){
   })
 
 
-  it("can be run with messages, by default", {
+  it("can be run with messages", {
     pkg <- local_package()
     p <- pkg_path(pkg)
     dir.create(file.path(p, "src"))
     file.copy(test_path("single.cpp"), file.path(p, "src", "single.cpp"))
-    expect_message(cpp_register(p), "1 functions decorated with [[cpp11::register]]", fixed = TRUE)
+    suppressMessages(
+      # cpp_register() sends 3 messages, but expect_message() only checks
+      # for one, so suppress the other ones, so that they don't appear in the
+      # testthat output
+      expect_message(
+        cpp_register(p, quiet = FALSE), "1 functions decorated with [[cpp11::register]]", fixed = TRUE)
+    )
```

Comment:
What if you just do `expect_snapshot()` to capture everything from `cpp11_register(p, quiet = FALSE)`?

Reply:
Thanks, that's much better. I've opened #291 to follow up for other use cases.